### ISSUE TITLE
notimplemented_placeholder flags abstract methods

### ISSUE THREAD

**Issue by rsionnach** [rsionnach/sloppylint#6]
## Issue
The `notimplemented_placeholder` pattern incorrectly flags abstract methods that raise `NotImplementedError`.

## Example
```python
from abc import ABC, abstractmethod

class MyBase(ABC):
    @abstractmethod
    def another_method(self):
        raise NotImplementedError  # FLAGGED - but this is valid!
```

## Expected Behavior
Methods decorated with `@abstractmethod` should not be flagged when raising `NotImplementedError`, as this is a standard Python pattern for abstract method definitions.

## Suggested Fix
Check for abstract decorators and Protocol/ABC class context before flagging.

**Comment by rsionnach** [comment]
This was already addressed. The `notimplemented_placeholder` pattern checks for:

1. **Abstract decorators**: `@abstractmethod`, `@abstractproperty`, `@abstractclassmethod`, `@abstractstaticmethod`, `@overload`
2. **Protocol/ABC context**: Methods inside classes inheriting from `Protocol`, `ABC`, `ABCMeta`, `Interface`

Methods with these decorators or in these contexts are NOT flagged when raising `NotImplementedError`.

Closing as fixed.
